### ISSUE TITLE
Don't know how to resolve variable in match in DBConnection.Connection.sync_stop/4

### ISSUE THREAD

**Issue by dudeman** [KronicDeth/intellij-elixir#392]
# Version

4.2.0

# Exception

## Message

Don't know how to resolve variable in match

### Excerpt

```
{:DOWN, ^mref, _, _, _} -> :ok
```
 Line(s) 431-431

## Stacktrace

```
org.elixir_lang.errorreport.Logger.error(Logger.java:48)
org.elixir_lang.errorreport.Logger.error(Logger.java:31)
org.elixir_lang.psi.scope.Variable.execute(Variable.java:136)
org.elixir_lang.psi.scope.Variable.execute(Variable.java:461)
org.elixir_lang.psi.scope.Variable.execute(Variable.java:68)
org.elixir_lang.psi.scope.Variable.execute(Variable.java:461)
org.elixir_lang.psi.scope.Variable.execute(Variable.java:68)
org.elixir_lang.psi.scope.Variable.execute(Variable.java:461)
org.elixir_lang.psi.scope.Variable.execute(Variable.java:68)
org.elixir_lang.psi.scope.Variable.execute(Variable.java:488)
org.elixir_lang.psi.scope.Variable.execute(Variable.java:477)
org.elixir_lang.psi.scope.Variable.execute(Variable.java:109)
org.elixir_lang.psi.scope.Variable.execute(Variable.java:320)
org.elixir_lang.psi.scope.Variable.execute(Variable.java:102)
org.elixir_lang.psi.impl.ElixirPsiImplUtil.processDeclarations(ElixirPsiImplUtil.java:1425)
org.elixir_lang.psi.impl.ElixirUnmatchedUnqualifiedNoParenthesesCallImpl.processDeclarations(ElixirUnmatchedUnqualifiedNoParenthesesCallImpl.java:121)
com.intellij.psi.util.PsiTreeUtil.treeWalkUp(PsiTreeUtil.java:1046)
org.elixir_lang.psi.scope.variable.MultiResolve.resolveResultList(MultiResolve.java:90)
org.elixir_lang.psi.scope.variable.MultiResolve.resolveResultList(MultiResolve.java:74)
org.elixir_lang.reference.Callable.multiResolve(Callable.java:782)
org.elixir_lang.reference.Callable.resolve(Callable.java:806)
org.elixir_lang.annonator.Callable$1.visitCall(Callable.java:60)
org.elixir_lang.annonator.Callable$1.visitElement(Callable.java:48)
com.intellij.psi.impl.PsiElementBase.accept(PsiElementBase.java:274)
org.elixir_lang.psi.impl.ElixirUnmatchedUnqualifiedNoArgumentsCallImpl.accept(ElixirUnmatchedUnqualifiedNoArgumentsCallImpl.java:39)
org.elixir_lang.annonator.Callable.annotate(Callable.java:39)
com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.runAnnotators(DefaultHighlightVisitor.java:139)
com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.visit(DefaultHighlightVisitor.java:102)
com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.runVisitors(GeneralHighlightingPass.java:334)
com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.lambda$collectHighlights$2(GeneralHighlightingPass.java:269)
com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.analyzeByVisitors(GeneralHighlightingPass.java:293)
com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.lambda$analyzeByVisitors$3(GeneralHighlightingPass.java:296)
com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.analyze(DefaultHighlightVisitor.java:86)
com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.analyzeByVisitors(GeneralHighlightingPass.java:296)
com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectHighlights(GeneralHighlightingPass.java:268)
com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectInformationWithProgress(GeneralHighlightingPass.java:207)
com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:82)
com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitorBasedInspection$MyPsiElementVisitor.visitFile(DefaultHighlightVisitorBasedInspection.java:182)
com.intellij.extapi.psi.PsiFileBase.accept(PsiFileBase.java:70)
com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitorBasedInspection.runGeneralHighlighting(DefaultHighlightVisitorBasedInspection.java:133)
com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitorBasedInspection.checkFile(DefaultHighlightVisitorBasedInspection.java:101)
com.intellij.codeInspection.ex.GlobalInspectionContextImpl.lambda$inspectFile$4(GlobalInspectionContextImpl.java:496)
com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:123)
com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1034)
com.intellij.concurrency.ApplierCompleter.lambda$wrapInReadActionAndIndicator$0(ApplierCompleter.java:96)
com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:494)
com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:443)
com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
com.intellij.concurrency.ApplierCompleter.wrapInReadActionAndIndicator(ApplierCompleter.java:106)
com.intellij.concurrency.ApplierCompleter.compute(ApplierCompleter.java:91)
java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
java.util.concurrent.ForkJoinPool$WorkQueue.pollAndExecCC(ForkJoinPool.java:1190)
java.util.concurrent.ForkJoinPool.helpComplete(ForkJoinPool.java:1879)
java.util.concurrent.ForkJoinPool.awaitJoin(ForkJoinPool.java:2045)
java.util.concurrent.ForkJoinTask.doJoin(ForkJoinTask.java:390)
java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:719)
java.util.concurrent.ForkJoinPool.invoke(ForkJoinPool.java:2616)
com.intellij.concurrency.JobLauncherImpl.invokeConcurrentlyUnderProgress(JobLauncherImpl.java:60)
com.intellij.concurrency.JobLauncher.invokeConcurrentlyUnderProgress(JobLauncher.java:57)
com.intellij.codeInspection.ex.GlobalInspectionContextImpl.inspectFile(GlobalInspectionContextImpl.java:492)
com.intellij.codeInspection.ex.GlobalInspectionContextImpl.lambda$null$2(GlobalInspectionContextImpl.java:417)
com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1034)
com.intellij.codeInspection.ex.GlobalInspectionContextImpl.lambda$runTools$3(GlobalInspectionContextImpl.java:413)
com.intellij.concurrency.JobLauncherImpl$1MyTask.lambda$call$0(JobLauncherImpl.java:284)
com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:494)
com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:443)
com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
com.intellij.concurrency.JobLauncherImpl$1MyTask.call(JobLauncherImpl.java:271)
com.intellij.concurrency.JobLauncherImpl$1MyTask.call(JobLauncherImpl.java:261)
java.util.concurrent.ForkJoinTask$AdaptedCallable.exec(ForkJoinTask.java:1424)
java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```

**Comment by KronicDeth** [comment]
I'm going to assume the problem is resolving the `mref`.  Can you post the entire both of the `receive` or `case` the surrouds the excerpted clause?

**Comment by KronicDeth** [comment]
I found the excerpt in https://github.com/elixir-ecto/db_connection/blob/790d41a51be9814ff1df1cd12896ccaae8c21f12/lib/db_connection/connection.ex#L82, so I'm going to use that as the test case.

**Comment by KronicDeth** [comment]
I can't reproduce with master ([703a150](https://github.com/KronicDeth/intellij-elixir/commit/703a150a7a897693c1c7dcacdcf2f7078b475de4)), but I can with [v4.2.0](https://github.com/KronicDeth/intellij-elixir/tree/v4.2.0).  Diffing the two, I think this was already fixed by #387, which is available in a canary build, [v4.3.0-dev+20160822081835](https://github.com/KronicDeth/intellij-elixir/releases/tag/v4.3.0-dev%2B20160822081835).
